### ISSUE TITLE
Proper __index writing and invocation on UserData

### ISSUE THREAD

**Issue by thrisp** [yuin/gopher-lua#125]
1. What version of GopherLua are you using? : 

whatever version I received using go get last week

2. What version of Go are you using? : 

go version go1.8.1

3. What operating system and processor architecture are you using? :
  
linux/amd64

I wrote an __index function internally in go to handle dot property notation

```
obj.property = value
print(obj.property) 
```

Without getting into the details yet, I am no longer able to access the obj methods 

```
obj:method()
obj.method()
```

without getting an error saying "attempt to call a non-function object"

If  I use something like and an __index: 

```
L.SetField(mt, "__index", L.SetFuncs(L.NewTable(), map[string]LGFunction))
```

I can access the methods but the situation is the reverse for attempts to access a property. 

I am not certain what to do interanlly to get both properties and methods on my UserData with an __index meta method custom written. 

So brief examples without a whole repo of code: 
 
```
func nodeIdx(t *Table, k string) l.LGFunction {
	return func(L *l.LState) int {
		return t.Execute(L)
	}
}

func (t *Table) Execute(L *l.LState) int {
	req := L.Get(2)
	k := req.String()
        // action is just a LGFunction
	if action, exists := t.accessible(k); exists {
		return action(L)
	}
	L.RaiseError("%s is not a member of %s or any table %s is descended from", k, t.name, t.name)
	return 0
}

func checkNode(L *l.LState, pos int) *node {
	ud := L.CheckUserData(pos)
	if v, ok := ud.Value.(*node); ok {
		return v
	}
	L.ArgError(pos, "node expected")
	return nil
}

func NewProperty(get, set l.LGFunction) l.LGFunction {
	return func(L *l.LState) int {
		switch L.GetTop() {
		case 2:
			if get != nil {
				return get(L)
			}
			L.RaiseError("cannot get %s property", L.Get(2))
		case 3:
			if set != nil {
				return set(L)
			}
			L.RaiseError("cannot set %s property", L.Get(2))
		default:
			L.RaiseError("error accessing property %s", L.Get(L.GetTop()))
		}
		return 0
	}
}

func NodeProperty(get, set nodeMemberFunc) l.LGFunction {
	return NewProperty(nodeMember(get), nodeMember(set))
}

type nodeMemberFunc func(*l.LState, *node) int

func nodeMember(fn nodeMemberFunc) l.LGFunction {
	return func(L *l.LState) int {
		if n := checkNode(L, 1); n != nil {
			return fn(L, n)
		}
		return 0
	}
}

func appendOut(L *l.LState, n *node) int {
	spew.Dump("APPEND")
	return 0
}

var nodeTable = &Table{
	lNodeClass,
	nil, // parent *Table
	map[string]MetaFunc{
		"__index":    nodeIdx,
	},
	map[string]l.LGFunction{
		"hidden":          NodeProperty(getHidden, setHidden),
		"append":         nodeMember(appendOut),
	},
}
```
tldr for the above 

```
n1 = node()
n1.hidden = false
print(n1.hiddden)
n2 = node()
n1:append(n2)
```

the part involving setting and getting the value of the property works fine, but when getting to append the program does spew out "APPEND" at the log, but then an error " attempt to call a non-function object" 

the above handles the properties ok as a starting point but errors on any method. I'm falsely assuming I can delegate everything to an LGFunction but thats not the case. Its naive and I guess I need to manipulate the stack more while doing this? I'm not sure what to do or why exactly I'm getting the error.

**Comment by thrisp** [comment]
Yes, I needed to push methods and resolve properties as separate acts.
